UseCases/Sources/HTTPEndToEnd/InstrumentedHTTPClient.swift

```diff
@@ -6,21 +6,36 @@ import NIO
 import NIOHTTP1
 
 struct InstrumentedHTTPClient {
-    private let client = HTTPClient(eventLoopGroupProvider: .createNew)
+    private let client: HTTPClient
     private let instrument: Instrument<HTTPHeaders, HTTPHeaders>
 
     init<I>(instrument: I, eventLoopGroupProvider: HTTPClient.EventLoopGroupProvider)
         where
         I: InstrumentProtocol,
         I.InjectInto == HTTPHeaders,
         I.ExtractFrom == HTTPHeaders {
+        self.client = HTTPClient(eventLoopGroupProvider: eventLoopGroupProvider)
         self.instrument = Instrument(instrument)
     }
 
+    // TODO: deadline: NIODeadline? would move into baggage?
```

Comment:
that's a good future question -- maybe ticket?

Maybe we'll make a fork of HTTPClient soon and propose how it'd look like with context?